test/e2e/set/scylladbcluster/multidatacenter/multidatacenter_scylladbcluster_config.go

```diff
@@ -0,0 +1,212 @@
+					CustomConfigSecretRef: pointer.Ptr(scyllaDBManagerAgentDatacenterRackTemplateSecret.Name),
+				},
+			}
+			sc.Spec.Datacenters[idx].Racks = []scyllav1alpha1.RackSpec{
+				{
+					Name: "a",
+					RackTemplate: scyllav1alpha1.RackTemplate{
+						ScyllaDB: &scyllav1alpha1.ScyllaDBTemplate{
+							CustomConfigMapRef: pointer.Ptr(scyllaDBDatacenterRackConfigMap.Name),
+						},
+						ScyllaDBManagerAgent: &scyllav1alpha1.ScyllaDBManagerAgentTemplate{
+							CustomConfigSecretRef: pointer.Ptr(scyllaDBManagerAgentDatacenterRackSecret.Name),
+						},
+					},
+				},
+			}
+		}
+
+		sc, err = metaCluster.ScyllaAdminClient().ScyllaV1alpha1().ScyllaDBClusters(userNS.Name).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaDBCluster %q roll out (RV=%s)", sc.Name, sc.ResourceVersion)
+		waitCtx2, waitCtx2Cancel := utils.ContextForMultiDatacenterScyllaDBClusterRollout(ctx, sc)
+		defer waitCtx2Cancel()
+		sc, err = controllerhelpers.WaitForScyllaDBClusterState(waitCtx2, metaCluster.ScyllaAdminClient().ScyllaV1alpha1().ScyllaDBClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, utils.IsScyllaDBClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		scylladbclusterverification.Verify(ctx, sc, rkcClusterMap)
+		err = scylladbclusterverification.WaitForFullQuorum(ctx, rkcClusterMap, sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, dc := range sc.Spec.Datacenters {
+			framework.By("Verifying if ConfigMaps and Secrets referenced by ScyllaDBCluster %q are mirrored in %q Datacenter", sc.Name, dc.Name)
+
+			clusterClient := rkcClusterMap[dc.RemoteKubernetesClusterName]
+
+			dcStatus, _, ok := slices.Find(sc.Status.Datacenters, func(dcStatus scyllav1alpha1.ScyllaDBClusterDatacenterStatus) bool {
+				return dc.Name == dcStatus.Name
+			})
+			o.Expect(ok).To(o.BeTrue())
+			o.Expect(dcStatus.RemoteNamespaceName).ToNot(o.BeNil())
+
+			for _, cm := range scyllaDBConfigMaps {
+				mirroredCM, err := clusterClient.KubeAdminClient().CoreV1().ConfigMaps(*dcStatus.RemoteNamespaceName).Get(ctx, cm.Name, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(mirroredCM.Data).To(o.Equal(cm.Data))
+			}
+
+			for _, secret := range scyllaDBManagerAgentSecrets {
+				mirroredSecret, err := clusterClient.KubeAdminClient().CoreV1().Secrets(*dcStatus.RemoteNamespaceName).Get(ctx, secret.Name, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				for k, v := range secret.Data {
+					decodedValue, err := io.ReadAll(base64.NewDecoder(base64.StdEncoding, bytes.NewReader(mirroredSecret.Data[k])))
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(mirroredSecret.Data).To(o.HaveKey(k))
+					o.Expect(decodedValue).To(o.Equal(v))
+				}
+			}
+
+			framework.By("Verifying if configuration of ScyllaDB Manager Agent is being used by %q datacenter", dc.Name)
```

Comment:
nit: this is a bit implicit with the auth token, maybe the comment could specify auth token propagation?